tvb_storage/tvb/storage/storage_interface.py

```diff
@@ -341,7 +346,66 @@ def move_datatype_with_sync(self, to_project, to_project_path, new_op_id, full_p
         self.sync_folders(to_project_path)
         self.set_project_inactive(to_project)
 
-        # Exporting related methods start here
+    # Method for preparing encryption before export
+
+    def prepare_encryption(self, user_public_key, project_name):
+        temp_folder = self.get_temp_folder(project_name)
+        public_key_file_name = "public_key_" + uuid.uuid4().hex + ".pem"
+        public_key_file_path = os.path.join(temp_folder, public_key_file_name)
+
+        if isinstance(user_public_key, FieldStorage) or isinstance(user_public_key, Part):
+            if not user_public_key.file:
+                return None, None
+
+            with open(public_key_file_path, 'wb') as file_obj:
+                self.copy_file(user_public_key.file, file_obj)
+        else:
+            shutil.copy2(user_public_key, public_key_file_path)
+
+        # Generate a random password for the files
+        pass_size = TvbProfile.current.hpc.CRYPT_PASS_SIZE
+        password = EncryptionHandler.generate_random_password(pass_size)
+
+        return public_key_file_path, password
+
+    # Methods related to ImportExportEncryptionHandler start here
+
+    def generate_public_private_key_pair(self, key_path):
+        self.import_export_encryption_handler.generate_public_private_key_pair(key_path)
+
+    def add_encrypted_suffix(self, name):
+        return self.import_export_encryption_handler.add_encrypted_suffix(name)
+
+    def get_path_to_encrypt(self, input_path):
+        return self.import_export_encryption_handler.get_path_to_encrypt(input_path)
+
+    @staticmethod
+    def load_public_key(public_key_path):
+        return ImportExportEncryptionHandler.load_public_key(public_key_path)
+
+    @staticmethod
+    def encrypt_password(public_key, symmetric_key):
+        return ImportExportEncryptionHandler.encrypt_password(public_key, symmetric_key)
+
+    def save_encrypted_password(self, encrypted_password, path_to_encrypted_password):
+        return self.import_export_encryption_handler.save_encrypted_password(encrypted_password,
+                                                                             path_to_encrypted_password)
+
+    def encrypt_and_save_password(self, public_key_path, password, path_to_encrypted_password):
+        self.import_export_encryption_handler.encrypt_and_save_password(public_key_path, password,
+                                                                        path_to_encrypted_password)
+
+    def __encrypt_data_at_export(self, file_path, password):
```

Comment:
same here

Reply:
Please review a bit all these new methods in StorageInterface to check which are not necessary to be exposed in the API